// interra-data-catalog-components/lib/components/Resource/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _ConfigurablePage = require('../ConfigurablePage');

var _ConfigurablePage2 = _interopRequireDefault(_ConfigurablePage);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Resource = function (_Component) {
    _inherits(Resource, _Component);

    function Resource() {
        _classCallCheck(this, Resource);

        return _possibleConstructorReturn(this, (Resource.__proto__ || Object.getPrototypeOf(Resource)).apply(this, arguments));
    }

    _createClass(Resource, [{
        key: 'render',
        value: function render() {

            var doc = this.props.doc;
            doc.format = doc.type;
            doc.accessURL = doc.uri;
            doc.preview = doc.uri;

            var configuration = {
                Main: {
                    accessURL: {
                        type: "FileDownload"
                    }
                },
                Table: {
                    format: {
                        type: "String",
                        label: "Format"
                    }
                }
            };

            return _react2.default.createElement(_ConfigurablePage2.default, { doc: doc, configuration: configuration });
        }
    }]);

    return Resource;
}(_react.Component);

exports.default = Resource;
