// interra-data-catalog-components/lib/components/PageSection/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _PageItemFileDownload = require('../PageItemFileDownload');

var _PageItemFileDownload2 = _interopRequireDefault(_PageItemFileDownload);

var _PageItemOrg = require('../PageItemOrg');

var _PageItemOrg2 = _interopRequireDefault(_PageItemOrg);

var _PageItemResource = require('../PageItemResource');

var _PageItemResource2 = _interopRequireDefault(_PageItemResource);

var _PageItemString = require('../PageItemString');

var _PageItemString2 = _interopRequireDefault(_PageItemString);

var _PageItemText = require('../PageItemText');

var _PageItemText2 = _interopRequireDefault(_PageItemText);

var _SectionTypeMain = require('../SectionTypeMain');

var _SectionTypeMain2 = _interopRequireDefault(_SectionTypeMain);

var _SectionTypeLeft = require('../SectionTypeLeft');

var _SectionTypeLeft2 = _interopRequireDefault(_SectionTypeLeft);

var _SectionTypeTable = require('../SectionTypeTable');

var _SectionTypeTable2 = _interopRequireDefault(_SectionTypeTable);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/*
import SectionTypeTitle from '../SectionTypeTitle';
import PageItemTheme from '../PageItemTheme';
import PageItemTag from '../PageItemTag';
import PageItemFilePreview from 'containers/PageItemFilePreview';
import PageItemImage from '../PageItemImage';
import PageItemExtentMap from '../PageItemExtentMap';
import PageItemSchema from '../PageItemSchema';
import PageItemSearchPage from '../PageItemSearchPage';
import PageItemSocial from '../PageItemSocial';
*/

var PageSection = function (_Component) {
    _inherits(PageSection, _Component);

    function PageSection() {
        _classCallCheck(this, PageSection);

        return _possibleConstructorReturn(this, (PageSection.__proto__ || Object.getPrototypeOf(PageSection)).apply(this, arguments));
    }

    _createClass(PageSection, [{
        key: 'render',
        value: function render() {

            var pageItems = {
                PageItemFileDownload: _PageItemFileDownload2.default,
                PageItemOrg: _PageItemOrg2.default,
                PageItemString: _PageItemString2.default,
                PageItemText: _PageItemText2.default,
                PageItemResource: _PageItemResource2.default
            };

            var SectionTypes = {
                SectionTypeMain: _SectionTypeMain2.default,
                SectionTypeLeft: _SectionTypeLeft2.default,
                SectionTypeTable: _SectionTypeTable2.default
            };

            var _props = this.props,
                doc = _props.doc,
                pageSchema = _props.pageSchema,
                type = _props.type;


            var SectionComponent = SectionTypes['SectionType' + type];

            var fields = pageSchema ? pageSchema[type] : {};

            if (fields && doc) {
                var section = Object.keys(fields).map(function (field, index) {
                    var Component = pageItems['PageItem' + fields[field].type];
                    var labelValue = 'label' in fields[field] ? fields[field].label : '';
                    var label = labelValue.length != 0;
                    var def = fields[field];
                    var item = {
                        data: {
                            field: field,
                            doc: doc,
                            def: def,
                            value: doc[field]
                        }
                    };
                    if (field in doc || field === 'widget') {
                        return _react2.default.createElement(Component, _extends({ labelValue: labelValue, label: label, key: index }, item));
                    }
                }, '<div></div>');
                return _react2.default.createElement(SectionComponent, section);
            } else {
                return _react2.default.createElement('span', null);
            }
        }
    }]);

    return PageSection;
}(_react.Component);

exports.default = PageSection;
