// interra-data-catalog-components/lib/components/FileDownload/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Wrapper = _interopRequireDefault(require("./Wrapper"));

var _FormatIcon = _interopRequireDefault(require("../FormatIcon"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function FileDownload(props) {
  var label = props.label,
      resource = props.resource;
  var format = resource.format.toUpperCase();

  var item = _react["default"].createElement("div", {
    className: "resource"
  }, _react["default"].createElement(_FormatIcon["default"], {
    format: format
  }), _react["default"].createElement("a", {
    href: resource.downloadURL,
    title: resource.format
  }, _react["default"].createElement("span", {
    "data-toggle": "tooltip",
    "data-placement": "top",
    "data-original-title": resource.format,
    "data-format": resource.format,
    className: "format-label"
  }, resource.format), resource.title));

  return _react["default"].createElement(_Wrapper["default"], {
    className: "file-download"
  }, label, " ", item);
}

FileDownload.propTypes = {
  item: _propTypes["default"].any,
  field: _propTypes["default"].any
};
var _default = FileDownload;
exports["default"] = _default;
